tool/generate_localization.dart

```diff
@@ -0,0 +1,199 @@
+    return (List<Locale> locales, Iterable<Locale> supported) {
+      if (locales == null || locales.isEmpty) {
+        return fallback ?? supported.first;
+      } else {
+        return _resolve(locales.first, fallback, supported, withCountry);
+      }
+    };
+  }
+
+  LocaleResolutionCallback resolution({Locale fallback, bool withCountry = true}) {
+    return (Locale locale, Iterable<Locale> supported) {
+      return _resolve(locale, fallback, supported, withCountry);
+    };
+  }
+
+  @override
+  Future<S> load(Locale locale) {
+    final String lang = getLang(locale);
+    if (lang != null) {
+      switch (lang) {
+""";
+
+const end = """
+        default:
+      }
+    }
+    S.current = const S();
+    return SynchronousFuture<S>(S.current);
+  }
+
+  @override
+  bool isSupported(Locale locale) => _isSupported(locale, true);
+
+  @override
+  bool shouldReload(GeneratedLocalizationsDelegate old) => false;
+
+  Locale _resolve(Locale locale, Locale fallback, Iterable<Locale> supported, bool withCountry) {
+    if (locale == null || !_isSupported(locale, withCountry)) {
+      return fallback ?? supported.first;
+    }
+
+    final Locale languageLocale = Locale(locale.languageCode, "");
+    if (supported.contains(locale)) {
+      return locale;
+    } else if (supported.contains(languageLocale)) {
+      return languageLocale;
+    } else {
+      final Locale fallbackLocale = fallback ?? supported.first;
+      return fallbackLocale;
+    }
+  }
+
+  bool _isSupported(Locale locale, bool withCountry) {
+    if (locale != null) {
+      for (Locale supportedLocale in supportedLocales) {
+        if (supportedLocale.languageCode != locale.languageCode) {
+          continue;
+        }
+        if (supportedLocale.countryCode == locale.countryCode) {
+          return true;
+        }
+        if (true != withCountry && (supportedLocale.countryCode == null || supportedLocale.countryCode.isEmpty)) {
+          return true;
+        }
+      }
+    }
+    return false;
+  }
+}
+
+String getLang(Locale l) => l == null
+  ? null
+  : l.countryCode != null && l.countryCode.isEmpty
+    ? l.languageCode
+    : l.toString();
+""";
+
+Future<void> main() async {
+  var inputContent = File(inputPath + 'strings_en.arb').readAsStringSync();
+  var config = json.decode(inputContent) as Map<String, dynamic>;
+  var output = '';
+
+  output += header;
+  output += textDirectionDeclaration;
+  output += localizedStrings(config: config, hasOverride: false);
+  output += '}' + '\n\n';
+
+  for (var locale in locales) {
+    output += 'class \$$locale extends S {' + '\n';
+    output += '  const \$$locale();' + '\n';
+
+    if (locale != locales.first) {
+      output += textDirectionDeclaration;
+
+      inputContent = File(inputPath + 'strings_$locale.arb').readAsStringSync();
+      config = json.decode(inputContent) as Map<String, dynamic>;
+
+      output += localizedStrings(config: config, hasOverride: true);
```

Comment:
Why here we have special case if we can try to generate `hasOverride` inside of the loop body ?